Documentation/Pipfile

```diff
@@ -6,10 +6,9 @@ verify_ssl = true
 [dev-packages]
 
 [packages]
-recommonmark = "*"
-m2r2 = "==0.2.5"
-Sphinx = "==3.2.1"
-sphinx-rtd-theme = "==0.5.0"
```

Comment:
We should be supplying minimum working versions for these. I'm sure there are versions that will not work.

I would rather just tell people using pipenv to just use pipenv install -r requirements.txt instead of maintaining things in two places. But I understand you want a fully pinned requirement list.

Reply:
Using requirements.txt was also my preference, but again I'm not a python expert

Reply:
What if we drop the Pipfile and leave the Pipfile.lock. This is a supported workflow for pipenv where it will then just auto create the Pipfile based on the requirements.txt there is even the `--ignore-pipfile` argument which will only use the lock file and not the Pipenv

Reply:
@btashton @v01d The problem with using just a requirements.txt file is that you have to manually update requirements, test, and pin them because you're trying to use one file for both functions; and you have no tooling to support you. In my experience this makes the project brittle. The requirements don't get updated often and the project breaks eventually. 

pipenv solves this by having a workflow that supports pinned requirements for CI and regular users, while developers can update the Pipfile.lock easily using pipenv's automatic pinning. I would prefer we keep them and do this instead whenever requirements are updated:

    pipenv run pip freeze > requirements.txt

That said, if both of you feel strongly that we should delete the Pipfile/Pipfile.lock I'll go along with it.

Reply:
My only expectation would be that both CI and people who clone the repo and tries to build the docs will use the same version of these packages and they will not be updated until someone decides to do so. If we stick to a version that works, I don't see how that could break. There's also not much reason to upgrade frequently. 

Anyway, I do not feel strongly about either, I don't have python experience to justify any preference. I don't like duplication in general and requirements.txt seems easier to me to maintain (as I don't use pyenv either). But I'll go with whatever you both feel is best.

Reply:
@v01d Ok, I would prefer to use the pipenv flow then. Re: requirements not changing... well I've been developing python in commercial environments for a long time. Every time my teams have frozen requirements and hoped they won't need to change, that bit us. Software on the Internet is bit like sand on the beach, it looks static but actually moves quite a bit over time.

In another PR I'll write the README.md that explains how to maintain the Pipenv, Pipfile.lock, and requirements.txt then.

Reply:
Ok, cool.

Remember that some of this information is already on the contributing -> documentation itself. Let's try to keep those with the same instructions. I'm thinking mostly about someone who just looks at the documentation and has not yet even cloned the NuttX repo.

Reply:
@v01d I'll keep them simple and make them match.